src/bpm/cgroups/cgroup.go

```diff
@@ -60,6 +69,80 @@ func Setup() error {
 	return nil
 }
 
+// subsystemGrouping fetches the parent cgroup grouping (if any) for a
+// particular subsystem.
+func subsystemGrouping(subsystem string) (string, error) {
+	// If the current process is not in a cgroup then we can do as we
+	// please. We do not mount cgroups together.
+	if _, err := os.Stat("/proc/self/cgroup"); os.IsNotExist(err) {
+		return subsystem, nil
+	}
+
+	// If the current process is in a cgroup then we need to match the
+	// grouping of the parent cgroup.
+	f, err := os.Open("/proc/self/cgroup")
```

Comment:
Is there a reason to do a separate `os.Stat` to check for file existence? `os.Open` should return an `os.IsNotExist` error right?